sbol_utilities/default_components.py

```diff
@@ -0,0 +1,148 @@
+
+    rbs = sbol3.Component(name, sbol3.SBO_DNA)
+    rbs.name = name
+    rbs.description = description
+    rbs.roles. append(sbol3.SO_RBS)
+    rbs.sequences.append(rbs_seq)
+    return tuple([rbs, rbs_seq])
+
+def cds(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a coding sequence (CDS) and a sequence Components with a given name, description, and sequence.
+    """
+    cds_seq = sbol3.Sequence(f'{name}_seq')
+    cds_seq.elements= sequence
+    cds_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    cds = sbol3.Component(name, sbol3.SBO_DNA)
+    cds.name = name
+    cds.description = description
+    cds.roles. append(sbol3.SO_CDS)
+    cds.sequences.append(cds_seq)
+    return tuple([cds, cds_seq])
+
+def terminator(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a terminator and a sequence Components with a given name, description, and sequence.
+    """
+    terminator_seq = sbol3.Sequence(f'{name}_seq')
+    terminator_seq.elements= sequence
+    terminator_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    terminator = sbol3.Component(name, sbol3.SBO_DNA)
+    terminator.name = name
+    terminator.description = description
+    terminator.roles. append(sbol3.SO_TERMINATOR)
+    terminator.sequences.append(terminator_seq)
+    return tuple([terminator, terminator_seq])
+
+def deg_tag(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a degradation tag and a sequence Components with a given name, description, and sequence.
+    """
+    deg_tag_seq = sbol3.Sequence(f'{name}_seq')
+    deg_tag_seq.elements= sequence
+    deg_tag_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    deg_tag = sbol3.Component(name, sbol3.SBO_DNA)
+    deg_tag.name = name
+    deg_tag.description = description
+    deg_tag.roles. append(tyto.SO.protein_stability_element)
+    deg_tag.sequences.append(deg_tag_seq)
+    return tuple([deg_tag, deg_tag_seq])
+
+def plasmid_vector(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a plasmid vector or backbone and a sequence Components with a given name, description, and sequence.
+    """
+    plasmid_vector_seq = sbol3.Sequence(f'{name}_seq')
+    plasmid_vector_seq.elements= sequence
+    plasmid_vector_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    plasmid_vector = sbol3.Component(name, sbol3.SBO_DNA)
+    plasmid_vector.name = name
+    plasmid_vector.description = description
+    plasmid_vector.roles. append(tyto.SO.plasmid_vector)
+    plasmid_vector.sequences.append(plasmid_vector_seq)
+    return tuple([plasmid_vector, plasmid_vector_seq])
+
+def engineered_region(name: str, description: str, sub_components: List[sbol3.SubComponent])-> sbol3.Component:
```

Comment:
What defines the set of the things that have been chosen for short-cut functions vs. things that do not have them?

Reply:
That is a very good question, I do it arbitrarily as they appear on my workflow but we could set a rational line, establish the things to be included and develop the package. My naive approximation is to start by all the things that we use in our workflows.

Reply:
I might suggest that everything that gets an explicit call-out in the specification is a good target, e.g., Table 4, the best practices sections I've referenced above, and other similar tables.

Reply:
I've added all the example on table 4 but the effector since it could be externally defined as well, more components can be added later.